tests/test_manager.py

```diff
@@ -221,3 +224,23 @@ async def test_lifespan_not_supported(app: typing.Callable) -> None:
     with pytest.raises(LifespanNotSupported):
         async with LifespanManager(app):
             pass  # pragma: no cover
+
+
+@pytest.mark.usefixtures("concurrency")
+async def test_lifespan_state_async_cm() -> None:
+    @contextlib.asynccontextmanager
+    async def lifespan(_app: ASGIApp) -> typing.AsyncGenerator:
+        yield {"foo": 1}
+
+    async def get(request: Request) -> Response:
+        assert request.state.foo == 1
+        request.state.foo = 2
+        return PlainTextResponse(f"Hello {request.state.foo}")
+
+    app = Starlette(lifespan=lifespan, routes=[Route("/get", get)])
+
+    async with LifespanManager(app) as lsm:
```

Comment:
This is a usage change that we should be mirroring in the documentation, I suppose?

I'm not sure if we want the usage to be:

```python
async with LifespanManager(app) as wrapped _app:
    ...
```

Or:

```python
from asgi_lifespan import LifespanManager, StateMiddleware

app = StateMiddleware(app)

async with LifespanManager(app):
    ...
```

The first option _looks_ a bit awkward. It's not clear what one should name the `wrapped_app` variable. It's also not obvious that `LifespanManager` should _yield_ anything, let alone a wrapped app?

The second option reduces the disruption on existing code, at the cost of requiring an additional explicit wrapping that users might not really know about. My understanding is that when using ASGITransport + Starlette, this would be mandatory, right?

Reply:
Hmm. Maybe something less surprising could be:

```python
async with LifespanManager(app) as manager:
    async with httpx.AsyncClient(app=manager.app) as client:
        ...
```